js/src/ExpectedMark.js

```diff
@@ -41,6 +41,27 @@ export default class ExpectedMark {
     this.onMarkListeners.push(onMark);
   }
 
+  /**
+   * This method tries to approximate full rendering lifecycle in the browser
+   * rather than just measuring JS execution like render() method does.
+   *
+   * See Nolan Lawson's article describing the issue and proposing this method:
+   * https://nolanlawson.com/2018/09/25/accurately-measuring-layout-on-the-web/
+   */
+  waitForNextPaintAndRecord() {
+    // In development mode, include DEBUG timestamps to show when
+    // original calls were fired to see the impact
+    if (
+      process.env.NODE_ENV !== "production" &&
+      typeof window.console !== "undefined" &&
+      typeof window.console.timeStamp !== "undefined"
+    ) {
+      window.console.timeStamp("[DEBUG] original call for " + this.name);
+    }
+
+    window.requestAnimationFrame(() => setTimeout(() => this.record()));
```

Comment:
trivial, but this can be slightly more concise

```js
window.requestAnimationFrame(() => setTimeout(this.record));
```

Reply:
Not really, setTimeout will not carry object binding with it so this is undefined inside record() call ;)

Reply:
ah right. In React, we often bind callback method like this in the constructor to avoid creating a new function each time the parent function runs

```js
this.record = this.record.bind(this); // guaranteed to keep the right `context` from now on...
```

no need to update unless you're squeezing out the last drop of performance. 👍

Reply:
I kind-of don't like the syntax - prefer class properties instead.

```javascript
mark = (name, waitForNextPaint = true) => {
...
}
```

Do you feel it's a bad use case for them or just to early to use them at this point?

Reply:
no, this is a perfect use case for it! and yeah, I like class property syntax better as well - you'll just need to add one more babel plugin to transpile them :)